### PR TITLE
Telemetry: Change TelemetryServletFilter getHttpRoute

### PR DESCRIPTION
**Pull request by Yas-Franco**
Fixes issue #26824

### CODE REVIEW

**Review by fmhwong — DISMISSED** [review]
LGTM

**Comment by Yas-Franco** [comment]
#build
#spawn.fullfat.buckets=io.openliberty.microprofile.telemetry.1.0.internal_fat,io.openliberty.microprofile.telemetry.1.0.internal.jaeger_fat,io.openliberty.microprofile.telemetry.1.0.internal_fat_tck,io.openliberty.microprofile.telemetry.1.1.internal_fat_tck

File: dev/io.openliberty.microprofile.telemetry.1.0.internal_fat/fat/src/io/openliberty/microprofile/telemetry/internal_fat/apps/servlet/HttpTraceTestServlet.java [hunk 1]
```diff
@@ -189,7 +215,7 @@ public void testStaticFile() throws Exception {
         assertThat(serverSpan, isSpan()
                         .withKind(SERVER)
                         .withAttribute(SemanticAttributes.HTTP_METHOD, "GET")
-                        .withAttribute(SemanticAttributes.HTTP_ROUTE, "/" + APP_NAME + "/" + HELLO_HTML)
+                        .withAttribute(SemanticAttributes.HTTP_ROUTE, "/" + APP_NAME)
                         .withAttribute(SemanticAttributes.HTTP_SCHEME, scheme)
```
Comment (Azquelt) [inline dev/io.openliberty.microprofile.telemetry.1.0.internal_fat/fat/src/io/openliberty/microprofile/telemetry/internal_fat/apps/servlet/HttpTraceTestServlet.java]: This looks good but can we add an assert for `HTTP_TARGET` here as well

File: dev/io.openliberty.microprofile.telemetry.1.0.internal/src/io/openliberty/microprofile/telemetry10/internal/rest/TelemetryContainerFilter.java [hunk 2]
```diff
@@ -173,6 +173,7 @@ public void filter(final ContainerRequestContext request) {
                     }
 
                     currentSpan.setAttribute(SemanticAttributes.HTTP_ROUTE, route);
+                    currentSpan.updateName(request.getMethod() + route);
```
Comment (Azquelt) [inline dev/io.openliberty.microprofile.telemetry.1.0.internal/src/io/openliberty/microprofile/telemetry10/internal/rest/TelemetryContainerFilter.java]: Does this pass all the existing tests? I'd have expected that you'd need to add a space between the method and the route.

**Comment by Azquelt** [comment]
I note that the `HttpSpanNameExtractorBuilder` which we're calling to compute the name in other places checks the `method` for being `null` and that it's listed in `HttpConstants.KNOWN_METHODS`. We should probably do the same unless we're sure it's not necessary.

**Comment by Yas-Franco** [comment]
#build
#spawn.fullfat.buckets=io.openliberty.microprofile.telemetry.1.0.internal_fat,io.openliberty.microprofile.telemetry.1.0.internal.jaeger_fat,io.openliberty.microprofile.telemetry.1.0.internal_fat_tck,io.openliberty.microprofile.telemetry.1.1.internal_fat_tck

File: dev/io.openliberty.microprofile.telemetry.1.1.internal/src/io/openliberty/microprofile/telemetry11/internal/rest/TelemetryContainerFilter.java [hunk 3]
```diff
@@ -164,7 +164,8 @@ public void filter(final ContainerRequestContext request) {
                     ROUTE_CACHE.putRoute(resourceClass, resourceMethod, route);
                 }
 
-                currentSpan.setAttribute(SemanticAttributes.HTTP_ROUTE, route);
+                currentSpan.setAttribute(SemanticAttributes.HTTP_ROUTE + " " + route);
+                currentSpan.updateName(request.getMethod() + route);
```
Comment (Azquelt) [inline dev/io.openliberty.microprofile.telemetry.1.1.internal/src/io/openliberty/microprofile/telemetry11/internal/rest/TelemetryContainerFilter.java]: I think you've added the space to the wrong line here.

**Review by Azquelt — APPROVED** [review]

**Comment by Yas-Franco** [comment]
#build
#spawn.fullfat.buckets=io.openliberty.microprofile.telemetry.1.0.internal_fat,io.openliberty.microprofile.telemetry.1.0.internal.jaeger_fat,io.openliberty.microprofile.telemetry.1.0.internal_fat_tck,io.openliberty.microprofile.telemetry.1.1.internal_fat_tck

**Comment by Yas-Franco** [comment]
#libby

**Pull request merged**